src/Editor/GUI/Editors/TriggerEditor.cpp

```diff
@@ -72,10 +72,11 @@ namespace GUI {
                     memcpy(name.data(), repeat->GetName().c_str(), std::min(repeat->GetName().size(), name.size()));
                     name[std::min(repeat->GetName().size(), name.size() - 1)] = '\0';
 
-
+                    std::string nameString;
+                    nameString = name.data();
                     // NAME
                     if (ImGui::InputText("Name", name.data(), name.size()))
-                        repeat->SetName(name.data());
+                        repeat->SetName(nameString);
```

Comment:
This won't work, right? We create `nameString` above and assign to it `name.data()`. This will copy that string, after which we modify `name` via ImGui. We then set `nameString` to `repeat` but it will not contain the new modified data. Since `SetName` takes a `std::string` it's fine to just pass `name.data()` directly as a `const char*` is implicitly converted to a `std::string`.

Reply:
Good point! However after I changed parameter on SetName from <std::string value> to <std::string& value> name.data() was no longer a valid input so I added nameString.